### DISCUSSION TITLE
Any way to use this with Google Gemini API?

### DISCUSSION THREAD

**Discussion by GoWithitRoger** [DeabLabs/cannoli#24] [Q&A] [answered]
This looks like a really powerful plugin. But I want to use Google Gemini, primarily for its free API (the workflows I have in mind are not privacy-sensitive). 

Can the plugin be adapt to use APIs other than OpenAI?

**Comment by cephalization** [comment]
@GoWithitRoger yes its possible, but not supported just yet. I have an open PR that would help make this possible, but I have gotten side tracked with other work and haven't gotten to merge it in yet. I will keep Gemini in mind and see what the API surface looks like before merging it #9

**Comment by cephalization** [comment]
This is in progress at #27 

Ollama is now supported, but not with function calling, which should be coming soon as well.

**Comment by cephalization** [comment, marked as answer]
Forgot to give an update here; Gemini is now officially supported! Let us know if you run into any issues
